version: '3.9'

services:
  rococo-db:
    container_name: rococo-db
    image: mysql:8.3.0
    environment:
      - MYSQL_ROOT_PASSWORD=secret
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      interval: 5s
      timeout: 5s
      retries: 5
    ports:
      - "3306:3306"
    networks:
      - rococo-network
    dns_search: .

  auth.rococo.dc:
    container_name: auth.rococo.dc
    image: ${PREFIX}/rococo-auth:latest
    ports:
      - "9000:9000"
    restart: always
    depends_on:
      rococo-db:
        condition: service_healthy
    networks:
      - rococo-network
    dns_search: .

  api.rococo.dc:
    container_name: api.rococo.dc
    image: ${PREFIX}/rococo-api:latest
    ports:
      - "8080:8080"
    restart: always
    depends_on:
      auth.rococo.dc:
        condition: service_started
    networks:
      - rococo-network
    dns_search: .

  client.rococo.dc:
    container_name: client.rococo.dc
    image: ${PREFIX}/rococo-client:latest
    ports:
      - "80:80"
    restart: always
    depends_on:
      api.rococo.dc:
        condition: service_started
    networks:
      - rococo-network
    dns_search: .

networks:
  rococo-network:
    driver: bridge
